Sources/Purchasing/StoreKitAbstractions/StoreKitWorkarounds.swift

```diff
@@ -75,3 +75,34 @@ extension SK1Product {
     }
 
 }
+
+extension SubscriptionPeriod {
+
+    /// This function simplifies large numbers of days into months and large numbers
+    /// of months into years if there are no leftover units after the conversion.
+    ///
+    /// Occassionally, StoreKit seems to send back a value 7 days for a 7day trial
+    /// instesad of a value of 1 week for a trial of 7 days in length.
+    /// Source: https://github.com/RevenueCat/react-native-purchases/issues/348
+    internal static func normalizeValueAndUnits(
+        value: Int,
+        unit: Unit
+    ) -> (value: Int, unit: Unit) {
```

Comment:
I don't think making this a mutating function on `SubscriptionPeriod` is possible since both `value` and `unit` are `let` constants. Additionally, I'm not sure if making this function's signature
```swift
internal static func normalize(_ subscriptionPeriod: SubscriptionPeriod) -> SubscriptionPeriod
```
is feasible either since the `normalize` function is called from within `SubscriptionPeriod`'s initializer, and such a call would lead to an endless call loop.

I'm not a fan of the function being static, but I think the only alternative is having the developer call `normalize` every time a `SubscriptionPeriod` is instantiated, which doesn't seem like a good design.

Reply:
Oh, I just saw [this comment](https://github.com/RevenueCat/purchases-ios/pull/1591#discussion_r878358928) 😅

I think it makes a lot of sense! I'll adopt that approach since this is just meant to be a StoreKit fix.